shared/middleware/net_sim_api.rb

```diff
@@ -18,33 +21,17 @@ class NetSimApi < Sinatra::Base
 
   end
 
-  # For test, make it possible to override the usual configured API choice
+  # For test, make it possible to override the usual configured API choices.
   @@overridden_pub_sub_api = nil
+  @@overridden_redis = nil
 
-  TableType = CDO.use_dynamo_tables ? DynamoTable : Table
-
-  def get_table(shard_id, table_name)
-    # Table name within channels API just concatenates shard + table
-    api_table_name = "#{shard_id}_#{table_name}"
-    TableType.new(CDO.netsim_api_publickey, nil, api_table_name)
-  end
-
-  # Get the Pub/Sub API interface for the current configuration
-  def get_pub_sub_api
-    return @@overridden_pub_sub_api unless @@overridden_pub_sub_api.nil?
-    CDO.use_pusher ? PusherApi : NullPubSubApi
-  end
-
-  # Set a particular Pub/Sub API interface to use - for use in tests.
-  #
-  # @param [PubSubApi] override_api
-  def self.override_pub_sub_api_for_test(override_api)
-    @@overridden_pub_sub_api = override_api
+  def initialize(app = nil)
+    super(app)
   end
 
-  def has_json_utf8_headers(request)
-    request.content_type.to_s.split(';').first == 'application/json' and
-        request.content_charset.to_s.downcase == 'utf-8'
+  # Return a new RedisTable instance for the given shard_id and table_name.
+  def get_table(shard_id, table_name)
+    RedisTable.new(get_redis_client, get_pub_sub_api, shard_id, table_name)
```

Comment:
I updated the code to give each shard its own key.  I was originally planning to do this in a follow-up PR but I realized it wasn't hard to do it right now. There are a number of tests to make sure that tables within a shard don't get intermingled.